examples/lighting-app/esp32/main/include/Button.h

```diff
@@ -15,16 +16,74 @@
  *    limitations under the License.
  */
 
+/**
+ * @file Button.h
+ *
+ * Describes a Button tied to a GPIO that provides debouncing
+ *
+ **/
+
 #pragma once
 
 #include "driver/gpio.h"
 #include "esp_log.h"
+#include "esp_system.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "freertos/timers.h"
 
+#define APP_BUTTON_PRESSED 0
+#define APP_BUTTON_RELEASED 1
+#define BUTTON_NUMBER 3
 class Button
 {
 public:
-    typedef void (*ButtonPressCallback)(void);
+    /**
+     * @brief
+     *  construct a button
+     */
+    Button();
```

Comment:
What's the point of providing the empty constructor? Won't it be useless if gpioNum isn't set?

Reply:
From reading this, it seems there is an Init that can set a gpio number. My guess is empty constructors allows for array definitions.

The interface is awkward though - gpio number should probably be optional (or nullable in some way) and there seems to be a conflict of `button() + init()` or `button(gpio) + init(gpio)` combinations not making sense.

Keeping somewhat as is (but I removed the redundant comments) to have this consistent with other places. Would be nice to have a saner interface though.